src/platform/native.zig

```diff
@@ -512,5 +522,19 @@ pub fn main() !void {
         }
 
         try app.update(&engine);
+
+        if (engine.internal.waitEventTimeout) |timeout| {
```

Comment:
I dont think this is the correct usage. From glfw docs, waitEvents works exactly like pollEvents except that it blocks until there is one or more events. So this whole section should be placed in if-else of the glfw.pollEvents() call above.

Reply:
I think I understand, let me try.